Doctrine/EmailUpdateListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace FOS\UserBundle\Doctrine;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use FOS\UserBundle\Model\UserInterface;
+
+
+/**
+ * Class EmailUpdateListener
+ * @package FOS\UserBundle\Doctrine
+ */
+class EmailUpdateListener implements EventSubscriber
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * Constructor
+     *
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    public function getSubscribedEvents()
+    {
+        return array(
+            'preUpdate',
+        );
+    }
+
+    /**
+     * Pre update listener based on doctrine common
+     *
+     * @param LifecycleEventArgs $args
+     */
+    public function preUpdate(LifecycleEventArgs $args)
+    {
+        $object = $args->getObject();
+        if ($object instanceof UserInterface) {
+
+            $emailUpdateConfirmation = $this->container->get('azine_platform.emailupdateconfirmation');
+
+            if($object->getConfirmationToken() != $emailUpdateConfirmation->getEmailConfirmedToken() && isset($args->getEntityChangeSet()['email'])){
+
+                $currentEmail = $args->getEntityChangeSet()['email'][0];
```

Comment:
IMHO $currentEmail is a poor choice for the variable name. Does it mean the old or the new one? We're just about to change it... I would use $oldEmail and $newEmail